src/core/lb.cpp

```diff
@@ -1254,10 +1263,13 @@ int lb_lbnode_get_pi_neq(int* ind, double* p_pi) {
 int lb_lbnode_get_boundary(int* ind, int* p_boundary) {
     if (lattice_switch & LATTICE_LB_GPU) {
 #ifdef LB_GPU
-        unsigned int host_flag;
+        int host_flag;
```

Comment:
This looks dubious. You declare the variable, but don't ever assign a value to it. So it has an undefined value, but you do an `if` below based on that value?